tensorflow_encrypted/protocol/pond.py

```diff
@@ -1454,6 +1505,64 @@ def _reveal_masked(prot, x):
     return prot.reveal(x.unmasked)
 
 
+#
+# im2col helpers
+#
+
+def _im2col_public(prot, x, h_filter, w_filter, padding, strides):
+    assert isinstance(x, PondPublicTensor), type(x)
+    x_on_0, x_on_1 = x.unwrapped
+
+    with tf.name_scope('im2col'):
+        with tf.device(prot.server_0.device_name):
+            x_on_0_col = x_on_0.im2col(h_filter, w_filter, padding, strides)
+
+        with tf.device(prot.server_1.device_name):
+            x_on_1_col = x_on_1.im2col(h_filter, w_filter, padding, strides)
+
+    x_col = PondPublicTensor(prot, x_on_0_col, x_on_1_col, x.is_scaled)
+    return x_col
+
+
+def _im2col_private(prot, x, h_filter, w_filter, padding, strides):
+    assert isinstance(x, PondPrivateTensor), type(x)
+    x_on_0, x_on_1 = x.unwrapped
+
+    with tf.name_scope('im2col'):
+        with tf.device(prot.server_0.device_name):
+            x_on_0_col = x_on_0.im2col(h_filter, w_filter, padding, strides)
+
+        with tf.device(prot.server_1.device_name):
+            x_on_1_col = x_on_1.im2col(h_filter, w_filter, padding, strides)
+
+    x_col = PondPrivateTensor(prot, x_on_0_col, x_on_1_col, x.is_scaled)
+
+    return x_col
+
+
+def _im2col_masked(prot, x_masked, h_filter, w_filter, padding, strides):
+    assert isinstance(x_masked, PondPublicTensor), type(x_masked)
+    a, a0, a1, alpha_on_0, alpha_on_1 = x_masked.unwrapped
+
+    with tf.name_scope('im2col'):
+
+        with tf.device(prot.crypto_producer.device_name):
+            a_col = a.im2col(h_filter, w_filter, padding, strides)
+
+        with tf.device(prot.server_0.device_name):
+            a0_col = a0.im2col(h_filter, w_filter, padding, strides)
+            alpha_on_0_col = alpha_on_0.im2col(h_filter, w_filter, padding, strides)
+
+        with tf.device(prot.server_1.device_name):
+            a1_col = a1.im2col(h_filter, w_filter, padding, strides)
+            alpha_on_1_col = alpha_on_1.im2col(h_filter, w_filter, padding, strides)
+
+    x_unmasked_col = prot.im2col(x_masked.unmasked, h_filter, w_filter, padding, strides)
```

Comment:
missing one indent here to get it under the `im2col` name scope (makes more sense in TensorBoard when they're nested)

Reply:
only this line correct? not the next one